# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Thu Sep 22 06:24:38 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Thu Sep 22 06:27:37 2022 +0000
@@ -667,11 +667,18 @@
   }
 
   /**
-   * Retruns true if we're in designMode.
+   * Return true if we're in designMode.
    */
   bool IsInDesignMode() const;
 
   /**
+   * Return true if entire the document is editable (although the document
+   * may have non-editable nodes, e.g.,
+   * <body contenteditable><div contenteditable="false"></div></body>
+   */
+  bool EntireDocumentIsEditable() const;
+
+  /**
    * Basically, this always returns true if we're for `contenteditable` or
    * `designMode` editor in web apps.  However, e.g., Composer of SeaMonkey
    * can make the editor not tabbable.